lib/rubocop/ast/processed_source.rb

```diff
@@ -159,6 +160,37 @@ def line_indentation(line_number)
           .length
       end
 
+      def node_tokens(node)
+        @node_tokens ||= {}.compare_by_identity
+        return @node_tokens[node] if @node_tokens[node]
+
+        begin_index = node_first_token_index(node)
+        end_index   = node_last_token_index(node)
+        @node_tokens[node] = tokens[begin_index..end_index]
+      end
+
+      def node_first_token(node)
+        node_tokens(node).first
```

Comment:
Are we sure we need the method at all? Implementation is not efficient...

Reply:
Changed the name and implementation; yes, we need it in several places, like https://github.com/rubocop-hq/rubocop/blob/30655006edf2921c323c9f7ec3883629dbde5179/lib/rubocop/cop/layout/space_inside_string_interpolation.rb#L58-L60 and https://github.com/rubocop-hq/rubocop/blob/30655006edf2921c323c9f7ec3883629dbde5179/lib/rubocop/cop/layout/space_inside_hash_literal_braces.rb#L97-L102